src/nav/ModalSearchNavBar.js

```diff
@@ -13,32 +12,26 @@ type Props = $ReadOnly<{|
   canGoBack: boolean,
 |}>;
 
-class ModalSearchNavBar extends PureComponent<Props> {
-  static contextType = ThemeContext;
-  context: ThemeData;
-
-  static defaultProps = {
-    canGoBack: true,
-  };
-
-  render() {
-    const { autoFocus, searchBarOnChange, canGoBack } = this.props;
-    return (
-      <View
-        style={{
-          borderColor: 'hsla(0, 0%, 50%, 0.25)',
-          flexDirection: 'row',
-          height: NAVBAR_SIZE,
-          alignItems: 'center',
-          borderBottomWidth: 1,
-          backgroundColor: this.context.backgroundColor,
-        }}
-      >
-        {canGoBack && <NavBarBackButton />}
-        <SearchInput autoFocus={autoFocus} onChangeText={searchBarOnChange} />
-      </View>
-    );
-  }
+export default function ModalSearchNavBar(props: Props) {
+  const { autoFocus, searchBarOnChange, canGoBack } = props;
+  const { backgroundColor } = useContext(ThemeContext);
+  return (
+    <View
+      style={{
+        borderColor: 'hsla(0, 0%, 50%, 0.25)',
+        flexDirection: 'row',
+        height: NAVBAR_SIZE,
+        alignItems: 'center',
+        borderBottomWidth: 1,
+        backgroundColor,
+      }}
+    >
+      {canGoBack && <NavBarBackButton />}
+      <SearchInput autoFocus={autoFocus} onChangeText={searchBarOnChange} />
+    </View>
+  );
 }
 
-export default ModalSearchNavBar;
+ModalSearchNavBar.defaultProps = {
+  canGoBack: true,
+};
```

Comment:
It looks like this is our first Hooks conversion involving defaultProps!

What do you think of handling this in one of the normal ways for just a function that takes some arguments? That seems like it'd be most in the spirit of Hooks. I think it'd also help make it clearer what's going on. As is, there's actually a bit of a regression in the readability of this code, I think, because the default (and the fact that this one *has* a default, and parent components aren't required to pass it) gets moved down here, far from the `Props` definition and the top of the function.

For example, we could say
```js
  const { autoFocus, searchBarOnChange, canGoBack = true } = props;
￼```
and then the `Props` type could say
```js
  canGoBack?: boolean,
```

That style would also have the added benefit that the `Props` type would now reflect the actual external interface. In the class-component world, we need to define a `Props` type that's inward-facing, reflecting the defaults already being applied, because the component's implementation may have a lot of entry points (the various methods that use `this.props`), and we need to express that they can all count on those props already being filled in, even if they're optional for callers. So we've accepted not actually having the interface written down anywhere, because the Props type is mostly close enough and the `defaultProps` are right there (at the top of the class, a few lines below the `Props` definition) and the duplication of writing both would be worse. But with a function component, there's just the one entry point (the top of the function), and we can handle filling in the defaults explicitly and do so just once.

Reply:
Sounds good!

Reply:
For cross-reference: I've now added this to the style guide:
https://github.com/zulip/zulip-mobile/blob/master/docs/style.md#react-function-prop-defaults
https://github.com/zulip/zulip-mobile/commit/316070448a9d168972c3be47121c1e51854df5bd

Reply:
Thank you!